Installation issue: qt-base@6 fails to build with gcc@7.5

### Steps to reproduce the issue

```console
$ spack install qt-base@6.3.2
...
```
After doing quick research on the internet, the build error shown below is possibly due to the version of gcc. Hence package file needs to improved by adding conflict statements.

### Error message

<details><summary>Error message</summary><pre>
...
/cache/xuser/spack-stage/spack-stage-qt-base-6.3.2-5gzbyvx2z3blriycfrapdrfimetdkjui/spack-src/src/corelib/text/qlocale_tools.cpp:57:10: fatal error: charconv: No such file or directory
 #include <charconv>
          ^~~~~~~~~~
compilation terminated.
...
</pre></details>


### Information on your system

```
$ spack debug report
* **Spack:** 0.20.0.dev0 (d29cb87ecca1c9270eefa6e77ec15b0546fa7458)
* **Python:** 3.6.8
* **Platform:** linux-centos7-haswell
* **Concretizer:** clingo
```

### Additional information

to be provided later. 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [ ] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/qt-base/package.py
+++ b/var/spack/repos/builtin/packages/qt-base/package.py
@@ -83,7 +83,7 @@ class QtBase(CMakePackage):
         depends_on("openssl")
 
     # Qt6 requires newer compilers: see https://github.com/spack/spack/issues/34418
-    conflicts('%gcc@:7')
+    conflicts("%gcc@:7")
 
     @property
     def archive_files(self):

--- a/var/spack/repos/builtin/packages/qt-base/package.py
+++ b/var/spack/repos/builtin/packages/qt-base/package.py
@@ -82,8 +82,7 @@ class QtBase(CMakePackage):
         depends_on("libproxy")
         depends_on("openssl")
 
-    # qt-base confirmed to build with gcc@8.5.0
-    # also see: https://doc.qt.io/qt-6/linux.html
+    # Qt6 requires newer compilers: see https://github.com/spack/spack/issues/34418
     conflicts('%gcc@:7')
 
     @property

--- a/var/spack/repos/builtin/packages/qt-base/package.py
+++ b/var/spack/repos/builtin/packages/qt-base/package.py
@@ -82,6 +82,10 @@ class QtBase(CMakePackage):
         depends_on("libproxy")
         depends_on("openssl")
 
+    # qt-base confirmed to build with gcc@8.5.0
+    # also see: https://doc.qt.io/qt-6/linux.html
+    conflicts('%gcc@:7')
+
     @property
     def archive_files(self):
         """Save both the CMakeCache and the config summary."""

